core/src/main/java/com/radixdlt/bootstrap/GenesisFileStore.java

```diff
@@ -69,76 +69,104 @@
 import com.radixdlt.utils.Compress;
 import com.radixdlt.utils.WrappedByteArray;
 import java.io.File;
-import java.io.FileInputStream;
 import java.io.FileOutputStream;
 import java.io.IOException;
 import java.nio.file.Files;
 import java.util.Optional;
+import org.bouncycastle.util.encoders.DecoderException;
+import org.bouncycastle.util.encoders.Hex;
 
 public final class GenesisFileStore implements GenesisStore {
   private static final int HASH_LENGTH = 32;
 
-  private final File file;
+  private final File genesisFolder;
+  private final File genesisDataFile;
+  private final File genesisHashFile;
 
-  public GenesisFileStore(File file) {
-    this.file = file;
+  public GenesisFileStore(File genesisFolder) {
+    this.genesisFolder = genesisFolder;
+    this.genesisDataFile = new File(genesisFolder, "genesis_data.bin");
+    this.genesisHashFile = new File(genesisFolder, "genesis_hash.txt");
   }
 
   @Override
   public void saveGenesisData(RawGenesisDataWithHash genesisDataWithHash) {
-    if (!file.exists()) {
+    if (!genesisFolder.exists()) {
       try {
-        final var unused = file.getParentFile().mkdirs();
-        if (!file.createNewFile()) {
-          throw new RuntimeException("Genesis file doesn't exist and failed to create");
+        if (!genesisFolder.mkdirs()) {
+          throw new RuntimeException("Genesis data folder doesn't exist and failed to create");
+        }
+        if (!genesisDataFile.createNewFile()) {
+          throw new RuntimeException("Genesis data file doesn't exist and failed to create");
+        }
+        if (!genesisHashFile.createNewFile()) {
+          throw new RuntimeException("Genesis hash file doesn't exist and failed to create");
```

Comment:
do they need to be explicitly created like that? won't `FileOutputStream` (re-)create them anyways?